File: LeetCode/Medium/3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.py
class Solution:
    def longestSubsequence(self, nums: List[int]) -> int:
        ma,xo = 0,0
        for i in nums:
            ma = max(ma,i)
            xo ^= i
        return 0 if ma == 0 else len(nums)-1 if xo == 0 else len(nums)

File: LeetCode/Medium/3702-longest-subsequence-with-non-zero-bitwise-xor/3702-longest-subsequence-with-non-zero-bitwise-xor.java
class Solution {
    public int longestSubsequence(int[] nums) {
        int ma = 0;
        int xo = 0;
        for(int i:nums){
            ma = Math.max(ma,i);
            xo ^= i;
        }
        if (ma == 0) return 0;
        if (xo == 0) return nums.length-1;
        return nums.length;
    }
}